Core/include/Acts/Geometry/GeometryIdentifierBlueprintNode.hpp

```diff
@@ -0,0 +1,110 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/BlueprintNode.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <memory>
+
+namespace Acts {
+
+class TrackingVolume;
+class PortalShellBase;
+class Volume;
+
+namespace Experimental {
+
+struct GeometryIdentifierBlueprintNodeImpl;
+
+/// @brief Blueprint node for configuring and applying geometry identifiers to volumes
+///
+/// This node must have exactly one child and applies geometry identifier
+/// configurations to the volumes in its subtree during finalization. Multiple
+/// configurations can be chained using the fluent interface.
+class GeometryIdentifierBlueprintNode : public BlueprintNode {
+ public:
+  /// Virtual destructor pushed to cpp file to avoid defining implementation
+  /// details
+  ~GeometryIdentifierBlueprintNode() override;
+
+  /// Default constructor
+  GeometryIdentifierBlueprintNode();
+
+  /// @brief Build the volume hierarchy under this node
+  /// @param options Blueprint build options
+  /// @param gctx The geometry context
+  /// @param logger The logger instance
+  /// @return Reference to the built volume
+  /// @note Requires exactly one child node
+  Volume& build(const BlueprintOptions& options, const GeometryContext& gctx,
+                const Logger& logger = Acts::getDummyLogger()) override;
+
+  /// @brief Connect portals in the volume hierarchy
+  /// @param options Blueprint build options
+  /// @param gctx The geometry context
+  /// @param logger The logger instance
+  /// @return Reference to the connected portal shell
+  PortalShellBase& connect(
```

Comment:
I missed this: why does the `GeometryIdentifierBlueprintNode` have a connect ? Is this just fulfilling the API?

Reply:
Yes, that's exactly the reason!